dask/dataframe/io/parquet.py

```diff
@@ -888,6 +889,8 @@ def to_parquet(df, path, engine='auto', compression='default', write_index=None,
 
     if compression != 'default':
         kwargs['compression'] = compression
+    elif 'snappy' in compress:
+        kwargs['compression'] = 'snappy'
```

Comment:
Apologies for the missed review, I'm (mostly) offline this week.

I'm not sure I agree with this change. I don't think defaults that depend on the environment you have installed are a good idea. In this case, whether the file is written with snappy compression depends on if snappy is installed, rather than erroring. This seems prone to error, especially since fastparquet doesn't depend on snappy being installed. Ideally both the fastparquet and arrow engines should produce (approximately) the same parquet output.

My preference would be the following:

- For one release cycle, issue a deprecation warning for `engine=='fastparquet'` if compression isn't manually specified, saying that in the future this will use snappy compression
- After that use snappy by default for fastparquet, erroring if not installed.

I like this better because it's clear from the code (and not the environment + code) what the output is, and produces consistent output across both backend engines.

Reply:
I'm curious, is this the same policy used by Pandas?

One of the benefits of using their defaults is that we get to avoid having to make all of these decisions :)

Reply:
As far as I understand from https://github.com/dask/dask/issues/3288, pandas uses snappy if available for either backend (and arrow defaults to it if used directly). I don't know what it does if snappy isn't around.

Reply:
Pandas passes `compression='snappy'` to both backends by default (it's the default value for the compression kwarg). The reason we have `compression='default'` currently was to work around fastparquet's lack of dependence on snappy.

My preference would be to make snappy the default for both engines, matching pandas' behavior. The proposal above was a way to do it with a deprecation period so it didn't break other people's code. A different way to achieve the same end goal would be to push a new release of fastparquet that requires snappy, keep this patch as is, and when dask requires fastparquet > the_new_version (which could be the next dask release, no strong thoughts), we can remove this logic and just have `compression='snappy'` as the kwarg instead of `compression='default'`.

Reply:
The only other major api difference is that dask's parquet engine defaults to `fastparquet`, whereas pandas' defaults to `pyarrow`. I'm not sure if we could/should change this one though.

Reply:
Part of the reason for preferring fastparquet has been that it has fewer dependencies, this is like compressions in dask.
Arrow-cpp seems to require all of the compressions to build, but actually I notice that only snappy is required for run with [condition](https://github.com/conda-forge/arrow-cpp-feedstock/blob/master/recipe/meta.yaml) `[win and py>=35]` - so what happens if you have arrow and not snappy?

Reply:
> Arrow-cpp seems to require all of the compressions to build, but actually I notice that only snappy is required for run with condition [win and py>=35] - so what happens if you have arrow and not snappy?

Snappy is built into the arrow c++ library, they don't depend on an external dependency here.

Reply:
FYI @jreback @wesm we're talking about Dask.dataframe's parquet compression defaults and how they relate to Pandas.

Reply:
To be clear, I don't think there's a question on current behavior, it's more if we should change this and how to go about doing it in a sane way. In summary:

- Pandas passes `compression='snappy'` by default to both engines.
- We currently pass `compression='snappy'` by default for pyarrow, and `compression=None` by default for fastparquet
- This PR changed this to pass `compression='snappy'` by default for fastparquet as well, but only if snappy is installed. This seems like a bad idea to me, as the behavior depends on the environment.
- I think we should be consistent with pandas here, and make `compression='snappy'` the hard default for both engines. The easiest way I can see to do this is make fastparquet depend on snappy. Dependencies these days are fairly cheap, and I'd argue that this behavior is more expected for users.

Reply:
Yup, I just ran into this issue. Didn't have snappy installed, and it created uncompressed files...  had to do the repartitioning again.

Reply:
@jglaser probably best to open a new issue at this point. Otherwise your comment is likely to be missed.